### ISSUE TITLE
Mismatch Jetpack Main Network Site Disables Auto-Updates

### ISSUE THREAD

**Issue by macmanx2** [Automattic/jetpack#1435]
If the Jetpack Main Network Site and Siteurl fields in Site Settings on the WordPress.com end do not match (even something as trivial as HTTP vs. HTTPS), the option to enable auto-updates disappears.

The bulk settings appear as:

![screen-shot-2014-12-16-at-3-29-23-pm](https://cloud.githubusercontent.com/assets/4856235/5464964/ba98a642-8544-11e4-8013-d9178b50fe35.png)

And, the individual settings appear as:

![screen-shot-2014-12-16-at-3-29-49-pm](https://cloud.githubusercontent.com/assets/4856235/5464967/c49a3764-8544-11e4-801c-a45e9bda1105.png)

Once the fields are edited to match, the option to enable auto-updates re-appears.

**Comment by macmanx2** [comment]
Just updating this ticket, the mismatch appears to re-occur every time Jetpack is updated.

**Comment by kraftbj** [comment]
@lezama - the option going away after each JP update is interesting. My personal site doesn't duplicate this, but it's an edge case of a site as it is.

**Comment by macmanx2** [comment]
Yeah, in my case it's happening at both http://macmanx.com/ and http://sarahwallinhuff.com/ but not http://delta.jhuff.me/

Both personal sites use admin over SSL, the test site does not.

**Comment by enejb** [comment]
There should be an explanation in calypso about it. This now. As to why you site can't update files. 
The problem is that http and https ( in most cases ) are the same site but they also could not be.

**Comment by macmanx2** [comment]
I'm conflicted in calling that a fix, since the administration over HTTPS instructions for WordPress do not mention changing this, and it has never been required before. In other words, I'm sure we'll have a lot of WordPress sites running admin over HTTPS with this problem.

The fact that no average user following https://codex.wordpress.org/Administration_Over_SSL will make this change, and that I've never seen an entirely different site operate over HTTPS vs. HTTP where WordPress is concerned, makes me feel like the real fix is to ignore HTTPS vs. HTTP when comparing the domain/directory.

**Comment by Ipstenu** [comment]
> There should be an explanation in calypso about it. This now. As to why you site can't update files. 

Sorry... what does that mean? Are you saying if we do admin over SSL, we should set siteurl to HTTPS?

That's going to cause a lot of complaints since we have, for ... god 10 years now? told people to use the define.

There's no way to push notices out to all the users to tell them to edit that. Also you'll be shooting Multisite in the foot since there is no difference between home and site URLs. If you have the define set in wp-config.php (like everyone would), then WordPress properly knows to use https.

I do agree that there's a situation where people might have one site for http and one for https, but with WordPress _that_ is the edge case. 80% of the users would not be doing that. 80% of the users would have the same site for http and https.

This should go the other way and have some opt-out or detection for the random 20%.

**Comment by roccotripaldi** [comment]
> Setting WordPress Address in Settings -> General (siteurl in the _options table) to HTTPS fixes this indeed, but I'm conflicted in calling that a fix, since the administration over HTTPS instructions for WordPress do not mention changing siteurl to HTTPS, and it has never been required before.

@Ipstenu @macmanx2 - i think we can safely tell users to do both. if they want to force ssl on the admin, the codex instructions still stand. if they want calypso to function properly, they'll need to update their settings.

here is where the comparison happens:
https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plugins/plugin-autoupdate-toggle/index.jsx#L66

one solution would be to disregard the protocol when making the comparision. but as pointed out above, the user may actually want two separate sites for http and https. that is their right, though I would never recommend that ;)

so in order for calypso to recognize their site at it's true URL, the settings should reflect the true URL.

**Comment by enejb** [comment]
>> There should be an explanation in calypso about it. This now. As to why you site can't update files.
> Sorry... what does that mean?
Sorry for my poor english. 

We have been always careful in Jetpack Manage to only allow sites to modify files if the site has the right permissions. One of the criteria that we look for is if the site_url() matches the network_site_url(). Which is intended to prevent sub sites that are on part of a multisite install from updating files. Currently it also catches sites that have https enabled for the admin site but not for front end. 

The problem with http and https that arises is that when we sync that data to .com if the user has define( 'FORCE_SSL_ADMIN', true ); is that even though the site_url should match the network_site_url for single/ main network sites it doesn't. See https://core.trac.wordpress.org/browser/tags/4.3.1/src/wp-includes/link-template.php#L2987

One solution that would work is if we took the `FORCE_SSL_ADMIN` into account. This solution would be to would require a new version of Jetpack since we currently don't sync that data.

The reason why we also check for if protocol matches is because it could be technically possible to have two sites running one on http and the other on https. ( Although very unlikely ) 

The reason why I think it would be a good idea to get user to update their site url to https in their settings ( Even though a hassle ) is
1. It would fix this issue for them. 
2. The site would be more secure for their visitors ( web wins ) 
3. Better google ranking http://googlewebmastercentral.blogspot.ca/2014/08/https-as-ranking-signal.html

cc @beaulebens and @mdawaffe for though and input. Should we ignore the protocol and allow sites to update files?

**Comment by macmanx2** [comment]
> so in order for calypso to recognize their site at it's true URL, the settings should reflect the true URL.

So, we're going to favor the 20% who use HTTPS in a weird way, over the 80% who have followed the official WordPress instructions which have stood that way for about 10 years?

When embracing WordPress's philosophy of "Decisions, not Options" that seems like the wrong direction, by a *very* large margin.

> One solution that would work is if we took the FORCE_SSL_ADMIN into account.

I'd be in favor of this, even if it required a new version. Anticipating users' needs FTW.

> The reason why I think it would be a good idea to get user to update their site url to https

Once huge downside is that HTTPS still breaks a lot of popular WordPress caching configurations, and bypasses all browser caching. Caching isn't needed in wp-admin, hence the existence of instructions to only secure that.

Basically, we'd be asking folks who are *already* securing wp-admin and have no reason to secure the front site, to take a huge performance hit just to have Jetpack Manage.

**Comment by Ipstenu** [comment]
I have to agree with @macmanx2 - users have a right to expect WordPress to work as it's worked and that Calypso would not require something that is _impossible_ for multisite.

@roccotripaldi I'm serious. I _cannot_ make the change in Multisite:

![screen shot 2015-11-30 at 2 09 07 pm](https://cloud.githubusercontent.com/assets/807903/11486137/f0735a2a-976b-11e5-96da-509f450ef71c.png)

So no, that is not an acceptable solution. I'm also willing to bet that more people use Multisite than use WordPress _and_ have a separate http and https site.

`FORCE_SSL_ADMIN` would make a lot more sense. That's what we tell Multisite folks to use to get https.

**Comment by roccotripaldi** [comment]
Sorry @Ipstenu - I didn't realize it was not possible to change the protocol of the main site's URL.
I think @enejb 's solution will work.

**Comment by kraftbj** [comment]
Marking as 3.8.2 wrt mocking an option for `FORCE_SSL_ADMIN` or however @enejb sees best to implement that :)

**Comment by enejb** [comment]
@macmanx2 and @Ipstenu: I completely agree that:
1. Users should expect things to work and not have to make any changes on their end.
2. That the likelihood of someone using multisite is far greater then have a separate http and https site.

Thanks for bringing this issue up.

**Comment by macmanx2** [comment]
Thanks, @enejb , looking forward to sharing the fix with our users. :)

**Comment by enejb** [comment]
This should be fixed with https://github.com/Automattic/wp-calypso/pull/1080 
@lezama pointed out that we do many more permissions checks when a user tries to update files. 

No update to Jetpack needed.
